fdbserver/include/fdbserver/workloads/TenantConsistency.actor.h

```diff
@@ -130,63 +118,108 @@ class TenantConsistencyCheck {
 			TenantGroupName tenantGroupName = t.getString(0);
 			int64_t tenantId = t.getInt(1);
 			ASSERT(self->metadata.tenantGroupMap.count(tenantGroupName));
-			ASSERT(self->metadata.tenantMap.count(tenantId));
+			ASSERT(localMap.count(tenantId));
 			self->metadata.tenantGroupIndex[tenantGroupName].insert(tenantId);
 			ASSERT(self->metadata.tenantsInTenantGroupIndex.insert(tenantId).second);
 		}
 		ASSERT_EQ(self->metadata.tenantGroupIndex.size(), self->metadata.tenantGroupMap.size());
 
-		return Void();
+		return localMap;
 	}
 
-	void validateTenantMetadata() {
-		if (metadata.clusterType == ClusterType::METACLUSTER_MANAGEMENT) {
-			ASSERT_LE(metadata.tenantMap.size(), metaclusterMaxTenants);
-		} else {
-			ASSERT_LE(metadata.tenantMap.size(), CLIENT_KNOBS->MAX_TENANTS_PER_CLUSTER);
-		}
-
-		ASSERT_EQ(metadata.tenantMap.size(), metadata.tenantCount);
+	void validateTenantMetadata(std::map<int64_t, TenantMapEntry> tenantMap) {
+		ASSERT(metadata.clusterType == ClusterType::METACLUSTER_DATA ||
+		       metadata.clusterType == ClusterType::STANDALONE);
+		ASSERT_LE(tenantMap.size(), CLIENT_KNOBS->MAX_TENANTS_PER_CLUSTER);
+		ASSERT_EQ(tenantMap.size(), metadata.tenantCount);
 		ASSERT_EQ(metadata.tenantNameIndex.size(), metadata.tenantCount);
 
-		int renameCount = 0;
-		for (auto [tenantId, tenantMapEntry] : metadata.tenantMap) {
+		for (auto [tenantId, tenantMapEntry] : tenantMap) {
 			ASSERT_EQ(tenantId, tenantMapEntry.id);
-			if (metadata.clusterType != ClusterType::METACLUSTER_DATA) {
+			// Only standalone clusters will have lastTenantId set
+			// For Metacluster, the lastTenantId field is updated for MetaclusterMetadata
+			// and not TenantMetadata
+			if (!isDataCluster) {
```

Comment:
Could this use the `clusterType` similar to the check that was there previously? If so, then I think we can avoid passing in and storing `isDataCluster`.